tests/h/search/query_test.py

```diff
@@ -97,12 +99,39 @@ def search(self, search):
 
 
 class TestGroupAuthFilter(object):
+    def test_unauthorized_user_can_not_see_the_groups_annotations(self,
```

Comment:
The content of these two tests looks fine, but I think the descriptions should be generalized:

- Annotations not in groups readable by the current user should not be returned by search
- Annotations in groups readable by the current user should be returned by search

These two general cases cover various concrete cases (eg. logged-out users not seeing annotations in private groups, logged-in users not seeing annotations in private groups they have not joined, logged-in users seeing annotations in private groups they have joined).